build.sbt

```diff
@@ -33,7 +33,7 @@ lazy val root = (project in file("."))
     description                   := "'Containerized exchange-api'",
     name                          := "amd64_exchange-api",
     organization                  := "org.openhorizon",
-    pekkoHttpVersion              := "[1.0.1]",
+    pekkoHttpVersion              := "[1.1.0]",
```

Comment:
You will also need to do a version bump to `1.125.1`. See .../resources/version.txt, /docs/*.json, and the CHANGELOG.md

Reply:
Ok. I will improve.

Reply:
Library pekko-http-jackson was updated to version 3.0.0. Warnings below, were fixed: 

```
[warn] /home/kevin/exchange-api/src/main/scala/org/openhorizon/exchangeapi/route/node/Node.scala:21:48: imported `Node` is permanently hidden by definition of trait Node in package nod
[warn] import org.openhorizon.exchangeapi.table.node.{Node, NodeRow, NodeType, NodesTQ}
[warn]                                                ^
[warn] /home/kevin/exchange-api/src/main/scala/org/openhorizon/exchangeapi/route/user/User.scala:15:48: imported `User` is permanently hidden by definition of trait User in package use
[warn] import org.openhorizon.exchangeapi.table.user.{User, UserRow, UsersTQ}
[warn]                                                ^
[warn] /home/kevin/exchange-api/src/main/scala/org/openhorizon/exchangeapi/route/agent/AgentConfigurationManagement.scala:66:72: slick.sql.FixedSqlStreamingAction[Seq[java.sql.Timestamp],java.sql.Timestamp,slick.dbio.Effect.Read] and Int are unrelated: they will most likely never compare equal
[warn]                     if(AgentVersionsChangedTQ.getChanged("IBM").result == 0)
[warn]                                                                        ^
[warn] three warnings found
```

Reply:
Unit-test warnings were fixed:
```
[warn] sbt server could not start because there's another instance of sbt running on this build.
[warn] Running multiple instances is unsupported
sbt:amd64_exchange-api> test
[info] compiling 69 Scala sources to /home/kevin/work/exchange-api/target/scala-2.13/test-classes ...
[warn] /home/kevin/work/exchange-api/src/test/scala/org/openhorizon/exchangeapi/route/organization/TestPostOrgChangesRoute.scala:327:5: A try without a catch or finally is equivalent to putting its body in a block; no exceptions are handled.
[warn]     try{
[warn]     ^
[warn] /home/kevin/work/exchange-api/src/test/scala/org/openhorizon/exchangeapi/route/user/TestPatchUserRoute.scala:182:22: Auto-application to `()` is deprecated. Supply the empty argument list `()` explicitly to invoke method updateUser,
[warn] or remove the empty argument list from its definition (Java-defined methods are exempt).
[warn] In Scala 3, an unapplied method like this will be eta-expanded into a function. [quickfixable]
[warn]         TESTUSERS(4).updateUser
[warn]    
```

Reply:
@naphelps I had a warning for part below. As I got we have never pass in first condition. I reworked it. Could you check?
```
if(AgentVersionsChangedTQ.getChanged("IBM").result == 0)
    AgentVersionsChangedTQ += (timestamp, "IBM")
 else
    AgentVersionsChangedTQ.getChanged("IBM").update(timestamp)
```

Reply:
@naphelps one more question, did I understand correctly that I need to update "**pekko-slf4j**" ?
`    "org.apache.pekko" %% "pekko-slf4j" % "[1.0.1]" `

Reply:
> @naphelps one more question, did I understand correctly that I need to update "**pekko-slf4j**" ? ` "org.apache.pekko" %% "pekko-slf4j" % "[1.0.1]"`

@omordyk I would like to hear your recommendation and justification first as an exercise.

Reply:
@naphelps . I will prepare recommendations. . Let me analyze dependency and warnings below:

```
[warn] Found version conflict(s) in library dependencies; some are suspected to be binary incompatible:
[warn] 	* com.fasterxml.jackson.module:jackson-module-scala_2.13:2.18.0 is selected over {2.16.0}
[warn] 	    +- com.github.pjfanning:pekko-http-jackson_2.13:3.0.0 (depends on 2.18.0)
[warn] 	    +- com.github.swagger-akka-http:swagger-scala-module_2.13:2.13.0 (depends on 2.18.0)
[warn] 	    +- com.github.swagger-akka-http:swagger-scala-module_2.13:2.12.0 (depends on 2.16.0)
[warn] 	    +- com.github.swagger-akka-http:swagger-pekko-http_2.13:2.12.0 (depends on 2.16.0)
[warn] 	* org.checkerframework:checker-qual:3.42.0 is selected over 2.8.1
[warn] 	    +- org.postgresql:postgresql:42.7.4                   (depends on 3.42.0)
[warn] 	    +- com.google.guava:guava:28.0-jre                    (depends on 2.8.1)
[warn] 	* org.slf4j:slf4j-api:2.0.13 is selected over {1.7.26, 2.0.9, 1.7.36}
[warn] 	    +- ch.qos.logback:logback-classic:1.5.6               (depends on 2.0.13)
[warn] 	    +- com.github.swagger-akka-http:swagger-scala-module_2.13:2.13.0 (depends on 2.0.13)
[warn] 	    +- org.apache.pekko:pekko-slf4j_2.13:1.0.1            (depends on 1.7.36)
[warn] 	    +- io.swagger.core.v3:swagger-core-jakarta:2.2.19     (depends on 2.0.9)
[warn] 	    +- com.typesafe.slick:slick_2.13:3.4.1                (depends on 1.7.36)
[warn] 	    +- com.github.swagger-akka-http:swagger-scala-module_2.13:2.12.0 (depends on 2.0.9)
[warn] 	    +- com.github.swagger-akka-http:swagger-pekko-http_2.13:2.12.0 (depends on 2.0.9)
[warn] 	    +- com.github.cb372:scalacache-core_2.13:0.28.0       (depends on 1.7.26)
```

Reply:
@naphelps Update from my side: 
- **pekko-http** was updated to version 1.1.0 to fix dependencies issue. Version 1.1.0 was required by **pekko-http-jackson** and others libs.
- **pekko-http**  depend on **pekko-http-cors**. As result **pekko-http-cors** was updated to 1.1.0
- Pekko modules, should have the same version to prevent  binary incompatibility. Also **pekko-slf4j** was updated to version 1.1.0.

- **pekko-http-jackson** was updated to include the latest (30 September)release and new fixes and features.

**The dependencies below were added:**

- **pekko-slf4** has compile dependency on **pekko-actor_3**.  
- **pekko-http-jackson** has compile dependency on **pekko-protobuf-v3_3**.
- **pekko-protobuf-v3_3** has compile dependency on **pekko-stream**.

Reply:
@naphelps I have a question regarding the [warning](https://github.com/open-horizon/exchange-api/pull/725#discussion_r1814396199) as I described above. Have these warnings been there before or should I fix it in this PR?

Reply:
@omordyk Excellent questions. The transitive dependencies through ScalaCache (https://github.com/cb372/scalacache) are responsible for the long standing dependency warnings that currently exist. Via slj4j-api, and checker-qual; see evicted for the conflicts. ScalaCache has multiple caching implementations underneath. We make use of the Google Guava implementation. The version used in the currently released prod version of scalacache contains a security vulnerability. ScalaCache as a library however is abandonware (see the github repo). The Exchange itself is very cache sensitive in performance, which Doug has sent years tuning. While the effort of moving to any other solution is minimal to moderate in terms of development lift, the tuning would have to start over from scratch. Our performance claims are a selling point, and we cannot have regressions in those claims.

Reply:
The above should be the last change. Everything else looks good.

Reply:
@naphelps I have removed theses dependencies:
```
      "org.apache.pekko" %% "pekko-actor" % "1.1.0",
      "org.apache.pekko" %% "pekko-stream" % "1.1.0",
      "org.apache.pekko" %% "pekko-protobuf-v3" % "1.1.0",
```
But I have an error after **stb reStart**:

```
root [2024-10-28 06:25:17,342] [org.apache.pekko.util.ManifestInfo]         [] [actors-pekko.actor.default-dispatcher-4]  [WARN]  -- You are using version 1.1.0 of Apache Pekko, but it appears you (perhaps indirectly) also depend on older versions of related artifacts. You can solve this by adding an explicit dependency on version 1.1.0 of the [pekko-protobuf-v3, pekko-stream] artifacts to your project. Here's a complete collection of detected artifacts: (1.0.1, [pekko-protobuf-v3, pekko-stream]), (1.1.0, [pekko-actor, pekko-slf4j]). See also: https://pekko.apache.org/docs/pekko/current/common/binary-compatibility-rules.html#mixed-versioning-is-not-allowed
root[ERROR] Exception in thread "main" java.lang.IllegalStateException: You are using version 1.1.0 of Apache Pekko, but it appears you (perhaps indirectly) also depend on older versions of related artifacts. You can solve this by adding an explicit dependency on version 1.1.0 of the [pekko-protobuf-v3, pekko-stream] artifacts to your project. Here's a complete collection of detected artifacts: (1.0.1, [pekko-protobuf-v3, pekko-stream]), (1.1.0, [pekko-actor, pekko-slf4j]). See also: https://pekko.apache.org/docs/pekko/current/common/binary-compatibility-rules.html#mixed-versioning-is-not-allowed
```
Attached a screen: 
<img width="1726" alt="Screenshot 2024-10-28 at 14 28 03" src="https://github.com/user-attachments/assets/da378a04-0d16-4047-94e6-e76ec50c674c">

Reply:
I guess they should be present

Reply:
@naphelps  **pekko-slf4j** was updated to version 1.1.1. **swagger-pekko-http**  was updated to version 2.14.0 to avoid adding a dependency **on pekko-actor, pekko-stream, pekko-protobuf.**

Reply:
LGTM